php_memcached_session.c

```diff
@@ -202,7 +196,7 @@ zend_bool s_configure_from_ini_values(memcached_st *memc, zend_bool silent)
 	}
 
 	if (MEMC_SESS_INI(consistent_hash_enabled)) {
-		check_set_behavior(MEMCACHED_BEHAVIOR_KETAMA, 1);
+		check_set_behavior(MEMC_SESS_INI(consistent_hash_type), 1);
```

Comment:
For some reason, the value of this macro is a randomly large number, when is should be "3" or "4". I haven't sorted out why it's not coming up correctly. printf in the OnUpdateConsistentHash function shows the expected value is set.

Reply:
Figured out that I was calling `OnUpdateString` with the integer field as the target, so the value was the actual pointer value to the string. Added an extra field to hold the string value, and a different field to hold the parsed integer value. Same pattern as the compression type INI setting.